### ISSUE TITLE
Parsing parent properties?

### ISSUE THREAD

**Issue by SeanOThomas** [cbeust/klaxon#153]
Klaxon can't seem to "see" properties within an abstract parent class. Is there a way around this?

**Comment by cbeust** [comment]
Can you include a failing test?

**Comment by SeanOThomas** [comment]
**FooBase and BarImpl**
```
abstract class FooBase(
        val id: String? = null
)

class BarImpl(
        val barValue: String? = null
): FooBase()
```
**Tests**
```
 on("parsing BarImpl") {

            val barImpl = Klaxon()
                    .parse<BarImpl>("""
                {
                    "id": "id123",
                    "barValue" : "value123"
                }
                """)

            it("should have value **passes**") {
                assertNotNull(barImpl?.barValue)
            }

            it("should have id **fails**") {
                assertNotNull(barImpl?.id)
            }
        }
```

**Comment by cbeust** [comment]
The problem here is that `id` is a `val`, so Klaxon cannot set it after creating the object.

Make it a `var` and your test will pass.

**Comment by cbeust** [comment]
Never mind, I found a workaround. Fixed in 3.0.5.

**Comment by darienalvarez** [comment]
@cbeust the problem persist on abstract classes if you try to annotate the attribute with @Json

`abstract class FooBase(
        @Json("_id")
        val id: String? = null
)

class BarImpl(
        val barValue: String? = null
): FooBase()`
